### PR TITLE
feat: SLA List Item component

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR includes,
1. Base settings list and list item components.
2. SLA list item component.

Fixes https://linear.app/chatwoot/issue/CW-3126/create-a-sla-list-item-component-with-the-new-design

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Screenshots**

**Desktop view**

| **Theme**        | **Preview** |
| ---                     | --- |
| **Light mode** | <image width="700" src="https://github.com/chatwoot/chatwoot/assets/64252451/7f73a604-9f94-4255-8bd0-a521a65a1949"></image> |
| **Dark mode** | <image width="700" src="https://github.com/chatwoot/chatwoot/assets/64252451/5cae55dc-bef0-4f8f-95b2-6d15b74d1885"></image> |

**Mobile view**

| **Light mode**  | **Dark mode** |
| ------------- | ------------- |
| Screenshot 2024-03-20 at 7 22 27 PM  | Screenshot 2024-03-20 at 7 22 01 PM |

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/settings/components/BaseSettingsList.vue [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<script setup></script>
+<template>
+  <div class="flex flex-col w-full h-full gap-3">
+    <slot />
+  </div>
+</template>
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/components/BaseSettingsList.vue:6]: Can this not be encapuslated in a Layout level component itself? 

SettingsLayout.vue
```vue
<template>
    <div>
        <slot for header>
        <slot for body>
    </div>
</template>
```

File: app/javascript/dashboard/routes/dashboard/settings/components/BaseSettingsListItem.vue [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        </p>
+      </div>
+    </slot>
+
+    <!-- right side section -->
+    <slot name="rightSection" />
+
+    <!-- actions section -->
+    <div
+      v-if="hasActions"
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/components/BaseSettingsListItem.vue:51]: See if we can check for `$slots.actions` is present?

File: app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAListItem.vue [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+</script>
+<template>
+  <base-settings-list-item
+    :title="slaName"
+    :description="description"
+    :has-actions="true"
+  >
+    <template #label>
+      <div
+        class="inline-flex items-center min-w-0 gap-1 px-1.5 sm:px-2 py-1 border border-solid rounded-lg border-slate-75 dark:border-slate-700/50"
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/sla/components/SLAListItem.vue:44]: See if this can be separated as a component?

**Review by scmmishra — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/routes/dashboard/settings/SettingsLayout.vue [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    -apple-system,
+    system-ui,
+    BlinkMacSystemFont,
+    'Segoe UI',
+    Roboto,
+    'Helvetica Neue',
+    Arial,
+    sans-serif !important;
+}
+</style>
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/SettingsLayout.vue:23]: You can add this to the tailwind config, that would be better. Or else, we can start using Inter everywhere? @pranavrajs what do you think?

Also @iamsivin see if we can use [fontsource.org](https://fontsource.org) for this... perhaps, we could use it. That will ensure we can update the font easily

File: app/javascript/dashboard/routes/dashboard/settings/components/BaseSettingsListItem.vue [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  },
+  description: {
+    type: String,
+    default: '',
+  },
+});
+</script>
+<template>
+  <div
+    class="flex relative flex-col sm:flex-row p-4 gap-4 sm:p-6 justify-between shadow-sm sm:divide-x sm:divide-slate-75 sm:dark:divide-slate-700/50 group bg-white border border-solid rounded-xl dark:bg-slate-800 border-slate-75 dark:border-slate-700/50 max-w-[900px] w-full"
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/components/BaseSettingsListItem.vue:15]: Why do we need the `divide` here?

File: app/javascript/dashboard/routes/dashboard/settings/sla/components/SLABusinessHoursLabel.vue [hunk 6]
```diff
@@ -0,0 +1,34 @@
+<script setup>
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/sla/components/SLABusinessHoursLabel.vue:1]: This is good for now. We can extract a base component from this, but that would be over-optimizing at this point. Let's keep it in mind though, the next time we build something similar, we can extract a base component out

**Review by scmmishra — COMMENTED** [review]
Minor comment, everything else is good to go. Great work on this @iamsivin 🖖

File: app/javascript/dashboard/routes/dashboard/settings/SettingsLayout.vue [hunk 7]
```diff
@@ -0,0 +1,23 @@
+<script setup></script>
```
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/SettingsLayout.vue:1]:
```suggestion
```

File: app/javascript/dashboard/routes/dashboard/settings/SettingsLayout.vue
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/SettingsLayout.vue:1]: Let's see if we can somehow merge this with `SettingsWrapper`

File: app/javascript/dashboard/routes/dashboard/settings/sla/components/SLABusinessHoursLabel.vue (hunk 6 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/settings/sla/components/SLABusinessHoursLabel.vue:1]: Yeah, sure

File: app/javascript/dashboard/routes/dashboard/settings/components/BaseSettingsListItem.vue (hunk 5 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/settings/components/BaseSettingsListItem.vue:15]: Its used to show the border between the left and the right section.
image

**Review by scmmishra — APPROVED** [review]

**Comment by scmmishra** [comment]
@sojan-official this is good to go, can you please merge this!

**Pull request merged**